### PR TITLE
Async

### PR DESCRIPTION
**Pull request by dodoburner**
I have done the following:
- Refactor the add and remove book action creators to return functions that update the api storage and the state
- Added an action creator that fetches the data and displays it on page load

### CODE REVIEW

File: src/redux/configureStore.js [hunk 1]
```diff
@@ -7,5 +8,8 @@ const rootReducer = combineReducers({
   categories: categoriesReducer,
 });
 
-const store = configureStore({ reducer: rootReducer });
+const store = configureStore({
+  reducer: rootReducer,
+  middleware: [thunk],
```
Comment (rloterh) [inline src/redux/configureStore.js:13]: - [ ] This module requires that you use `createAsyncThunk` for creating _thunk action creators_, which should be similar to this example:

```javascript
import { createAsyncThunk, createSlice } from '@reduxjs/toolkit';
import axios from 'axios'; 


export const getBooks = createAsyncThunk(
  'books/getBooks',
  async () => {
    const response = await axios.get(
      'https://us-central1-bookstore-api-e63c8.cloudfunctions.net/bookstoreApi/',
    );
    return response.data;
  },
); 
```

**Review by rloterh — CHANGES_REQUESTED** [review]
Hi Dorian, 

You did extremely great on this project! 👏 👏 
There are however a few issues that you still need to work on to go to the next project but you are almost there! 🚦 🚦 🚦  

## Highlights 👍  
- Nice job for using axios for API calls


## Required Changes ♻️
_Kindly check the comments under the review._


<br>
Cheers and Happy coding!👏 💻 🍷

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/redux/configureStore.js (hunk 1 above)
Comment (dodoburner) [reply src/redux/configureStore.js:13]: I think this is wrong, I have explained in the PR description that it is not possible without refactoring everything with createSplice, which was not how Microverse instructed us in the first place. The example that you provided does not show how to use it with the current configuration it only shows how to create a thunk with createAsyncThunk and not how to use it. @rloterh

Comment (rloterh) [reply src/redux/configureStore.js:13]: Kindly check the example link in slack. Kindly read further on the same [here](https://redux-toolkit.js.org/api/createAsyncThunk)

Cheers!

Comment (dodoburner) [reply src/redux/configureStore.js:13]: Thank you very much for your help @rloterh.

**Review by danmainah — APPROVED** [review]
Hi @dodoburner, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 🎉 🎉 

## Highlights
✔️ No linter errors
✔️ Gitflow was correctly used

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**